core/store/src/flat/manager.rs

```diff
@@ -156,4 +156,11 @@ impl FlatStorageManager {
 
         Ok(())
     }
+
+    pub fn toggle_flat_state_updates(&self, enabled: bool) {
```

Comment:
Nit: `toggle_flat_state_updates` makes me think that we flip the state here which is not the case. I suggest renaming it to `set_flat_state_updates_mode` and `set_flat_head_update_mode`, respectively.

It's possible that this logic will stay in our code, because we may want to pause flat state updates during making checkpoint for state sync.

Reply:
yeah, that is a very good point!